src/base/utils/fs.cpp

```diff
@@ -128,17 +132,26 @@ bool Utils::Fs::smartRemoveEmptyFolderTree(const QString &path)
               , [](const QString &l, const QString &r) { return l.count('/') > r.count('/'); });
 
     for (const QString &p : asConst(dirList)) {
-        // remove unwanted files
-        for (const QString &f : deleteFilesList) {
-            forceRemove(p + f);
-        }
-
-        // remove temp files on linux (file ends with '~'), e.g. `filename~`
         const QDir dir(p);
+        if (!dir.isEmpty(QDir::AllDirs | QDir::NoDotAndDotDot))
+            continue;
+
         const QStringList tmpFileList = dir.entryList(QDir::Files);
+        // deleteFilesList contains unwanted files, usually created by the OS
+        // temp files on linux usually end with '~', e.g. `filename~`
+        const auto predicate = [&deleteFilesList](const QString &f)
+        {
+            return (deleteFilesList.contains(f, Qt::CaseInsensitive)
+                    || f.endsWith('~'));
+        };
+        if (!std::all_of(tmpFileList.cbegin(), tmpFileList.cend(), predicate))
```

Comment:
I would store it to a const bool variable instead. The predicate lambda itself doesn't mean much but the result of std::all_of does.

Reply:
The return value is used only in that if statement. Does it matter if it is stored or not?
Unless, you mean that it improves readability, which I don't think so.

Reply:
It improves readability (for me) and debuggability (that you can type in variable name to watch).

Reply:
```c++
const bool hasOtherFiles = !std::all_of(tmpFileList.cbegin(), tmpFileList.cend(), []()
{
    // ...
});
```

Reply:
Done.